packages/wrangler/src/pipelines/index.ts

```diff
@@ -17,26 +16,17 @@ import {
 	updatePipeline,
 } from "./client";
 import type { CommonYargsArgv, CommonYargsOptions } from "../yargs-types";
-import type { PipelineUserConfig } from "./client";
-import type { Argv } from "yargs";
-
-// flag to skip delays for tests
-let __testSkipDelaysFlag = false;
+import type { Pipeline, PipelineUserConfig } from "./client";
+import type yargs from "yargs";
 
 async function authorizeR2Bucket(
 	name: string,
 	accountId: string,
-	bucket: string
+	destination: Pipeline["destination"]
 ) {
-	try {
-		await getR2Bucket(accountId, bucket);
-	} catch (err) {
-		if (err instanceof APIError) {
-			if (err.code == 10006) {
-				throw new FatalError(`The R2 bucket [${bucket}] doesn't exist`);
-			}
-		}
-		throw err;
+	const bucket = destination.path.bucket;
+	if (!(await getR2Bucket(accountId, bucket))) {
+		logger.warn("Specified bucket doesn't exist:", bucket);
```

Comment:
Why is this no longer a fatal error?
Is it valid to use a non-existing bucket?

Reply:
That crept in from a previous commit. I've reverted the change back to Oliver's original now.